components/in_memory_engine/src/config.rs

```diff
@@ -1,20 +1,160 @@
-use std::sync::Arc;
+use std::{error::Error, sync::Arc, time::Duration};
 
 use online_config::{ConfigChange, ConfigManager, OnlineConfig};
-use tikv_util::{config::VersionTrack, info};
+use serde::{Deserialize, Serialize};
+use tikv_util::{
+    config::{ReadableDuration, ReadableSize, VersionTrack},
+    info,
+};
 
-use crate::InMemoryEngineConfig;
+const DEFAULT_GC_RUN_INTERVAL: Duration = Duration::from_secs(180);
+// The minimum interval for GC run is 10 seconds. Shorter interval is not
+// meaningful because the GC process is CPU intensive and may not complete in
+// 10 seconds.
+const MIN_GC_RUN_INTERVAL: Duration = Duration::from_secs(10);
+// The maximum interval for GC run is 10 minutes which equals to the minimum
+// value of TiDB GC lifetime.
+const MAX_GC_RUN_INTERVAL: Duration = Duration::from_secs(600);
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, OnlineConfig)]
+#[serde(default, rename_all = "kebab-case")]
+pub struct InMemoryEngineConfig {
+    /// Determines whether to enable the in memory engine feature.
+    pub enable: bool,
+    /// The maximum memory usage of the engine.
+    pub capacity: Option<ReadableSize>,
+    /// When memory usage reaches this amount, we start to pick some regions to
+    /// evict.
+    pub evict_threshold: Option<ReadableSize>,
+    /// When memory usage reaches this amount, we stop loading regions.
+    // TODO(SpadeA): ultimately we only expose one memory limit to user.
+    // When memory usage reaches this amount, no further load will be
+    // performed.
+    pub stop_load_threshold: Option<ReadableSize>,
+    /// Determines the oldest timestamp (approximately, now - gc_run_interval)
+    /// of the read request the in memory engine can serve.
+    pub gc_run_interval: ReadableDuration,
+    pub load_evict_interval: ReadableDuration,
+    pub expected_region_size: Option<ReadableSize>,
```

Comment:
This field is used to determine the region size, thus why not directly use `raftstore.coprocessor.region_split_size` instead? /cc @SpadeA-Tang